alacritty_terminal/src/keyboard_motion.rs

```diff
@@ -0,0 +1,767 @@
+use std::cmp::min;
+
+use serde::Deserialize;
+
+use crate::event::EventListener;
+use crate::grid::{GridCell, Scroll};
+use crate::index::{Column, Point};
+use crate::term::cell::Flags;
+use crate::term::{Search, Term};
+
+/// Possible keyboard motion movements.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Deserialize)]
+pub enum KeyboardMotion {
+    /// Move up.
+    Up,
+    /// Move down.
+    Down,
+    /// Move left.
+    Left,
+    /// Move right.
+    Right,
+    /// Move to start of line.
+    First,
+    /// Move to end of line.
+    Last,
+    /// Move to the first non-empty cell.
+    FirstOccupied,
+    /// Move to top of screen.
+    High,
+    /// Move to center of screen.
+    Middle,
+    /// Move to bottom of screen.
+    Low,
```

Comment:
I think I like the names in the comments more TBH. Moving `High` is much more confusing than moving `Top`.

Reply:
`High` is much more in line with vim's naming though and matches the `H`/`M`/`L` of the default bindings.

Top/Bottom is also for the scrollback buffer, instead of just the visible region.

So I definitely think it would be a massive mistake to change this.